doc/src/BCMPackage.rst

```diff
@@ -30,6 +42,10 @@ Sets the version of the package.
 
 This will parse the version from a header file. It will parse the macros defined in the header as ``<prefix>_<package-name>_VERSION_MAJOR``, ``<prefix>_<package-name>_VERSION_MINOR``, and ``<prefix>_<package-name>_VERSION_PATCH``. The ``prefix`` by default is the package name in all caps, but can be set using the ``VERSION_PREFIX`` option.
 
+.. I'm a bit unsure about the above and following option. I think many codebases
+   use their own "version header" format and hardcoding this one is a but
+   unflexible.
```

Comment:
Yes, but if a project uses a different version header format, then they can parse the header themselves and set the header:

```cmake
# Parse custom version header
parse_header_version_header(MYVERSION)
bcm_setup_version(VERSION ${MYVERSION})
```

This format is what is used by some boost projects and kde projects as well(they provided similar functionality in their cmake modules: [ecm_setup_version](https://api.kde.org/ecm/module/ECMSetupVersion.html)).